emodelrunner/protocols/reader.py

```diff
@@ -0,0 +1,691 @@
+            "RampProtocol", "RatSSCxThresholdDetectionProtocol", "Vecstim", "Netstim"]
+        # fmt: on
+
+        for protocol_name, protocol_definition in protocol_definitions.items():
+            if protocol_name not in ["Main", "RinHoldcurrent"]:
+                recordings = get_recordings(
+                    protocol_name, protocol_definition, prefix, apical_point_isec
+                )
+
+                if "type" in protocol_definition:
+                    # add protocol to protocol dict
+                    self._parse_step_and_ramp(
+                        protocol_definition,
+                        protocol_name,
+                        sscx_protocols,
+                        recordings,
+                        stochkv_det,
+                    )
+                    self._parse_sscx_threshold_detection(
+                        protocol_definition, recordings, prefix
+                    )
+                    self._parse_vecstim_netstim(
+                        protocol_definition, protocol_name, recordings, syn_locs
+                    )
+
+                elif (
+                    "type" not in protocol_definition
+                    or protocol_definition["type"] not in protocols_with_type
+                ):
+                    stimuli = []
+                    for stimulus_definition in protocol_definition["stimuli"]:
+                        stimuli.append(
+                            ephys.stimuli.NrnSquarePulse(
+                                step_amplitude=stimulus_definition["amp"],
+                                step_delay=stimulus_definition["delay"],
+                                step_duration=stimulus_definition["duration"],
+                                location=SOMA_LOC,
+                                total_duration=stimulus_definition["totduration"],
+                            )
+                        )
+
+                    self.protocols_dict[protocol_name] = ephys.protocols.SweepProtocol(
+                        name=protocol_name, stimuli=stimuli, recordings=recordings
+                    )
+
+        if "Main" in protocol_definitions.keys():
+            self._parse_sscx_main(protocol_definitions, prefix)
+        else:
+            check_for_forbidden_protocol(self.protocols_dict)
+
+        return self.protocols_dict
+
+    def parse_thalamus_protocols(
+        self,
+        protocols_filepath,
+        stochkv_det=None,
+        prefix="",
+    ):
+        """Parses the Thalamus protocols from the json file input.
+
+        Args:
+            protocols_filename (str): path to the protocols file
+            stochkv_det (bool): set if stochastic or deterministic
+            prefix (str): prefix used in naming responses, features, recordings, etc.
+
+        Returns:
+            dict containing the protocols
+        """
+        protocol_definitions = self.load_protocol_json(protocols_filepath)
+
+        # fmt: off
```

Comment:
It's black, not me :)
Black is sometimes invasive maybe for the future we can discuss its alternatives.

Reply:
Haha, np.
I personally don't mind black, but if you want to switch to an alternative, we can discuss it.

Reply:
I'm overall happy with a strict formatter like black because it allows me to search (using `grep` for example) the entire code base for small code pieces e.g. `gid = config.getint("Cell", "gid")` 
With black, I know that there is always going to be a space after the comma so I can safely assume my search will return every usage of the code I'm looking for.